### ISSUE TITLE
Chapter 5 - Story of Lamech 

### ISSUE THREAD

**Issue by cgwid** [markjprice/cs11dotnet7#59]
I looked through the open and closed issue titles several times and did a search but didn't see anything related. I also reviewed my code several times to see if I made a mistake somewhere but I don't see one so I think this is an issue.

Chapter: 5

Page Number: 238 of pdf

Section Title: Implementing functionality using methods

Step Number: 2, 4, 5

Problem to fix: 
In step 4 we set up Lamech to get married twice:

<img width="591" alt="Screenshot 2023-04-12 at 3 54 15 PM" src="https://user-images.githubusercontent.com/12521724/231594364-3059ca98-aabb-4be7-b39c-4da135330786.png">

However, the instance method "Marry" defined in step 2 checks to see if the person is already married which breaks out of the recursion but also prevents subsequent marriages from occurring - "if (married) return":

<img width="576" alt="Screenshot 2023-04-12 at 3 52 25 PM" src="https://user-images.githubusercontent.com/12521724/231594089-50cf208c-ddea-464b-a8d6-7f8256775343.png">

So in step 4 when we try to marry zillah and lamech, it doesn't work. Then later in step 4 we call the "Procreate" method between the 2 but they aren't spouses so it triggers the exception:

<img width="664" alt="Screenshot 2023-04-12 at 3 53 40 PM" src="https://user-images.githubusercontent.com/12521724/231594278-98e34ea2-a5a0-4e2b-b588-39c6e690fe0b.png">

Output from vscode terminal:
<img width="569" alt="Screenshot 2023-04-12 at 3 56 44 PM" src="https://user-images.githubusercontent.com/12521724/231594741-5d9fd2fb-33f0-4651-9340-d44ad6d573a3.png">

vs.

Output in book:
<img width="496" alt="Screenshot 2023-04-12 at 3 59 36 PM" src="https://user-images.githubusercontent.com/12521724/231595162-1475b4cc-ca8a-4f8f-a0c8-66405196b078.png">

Suggested solution:
Seems there could be various ways to "fix" this or restructure but here is what I came up with: 

1) Have the spouse field be a List\<Person\>

2) Remove the married check in "Marry" method. Instead, check if partner name is in Spouse list and return to break out of the recursion. 

3) In "Procreate" method, check that the spouse is in the list before continuing.

4) Created a "DisplaySpouses" static method that takes in a person and checks their Spouse list to write the names to the console

**Comment by markjprice** [comment]
You are correct. Although I could solve this issue in a way similar to how you suggest, I need some time to think through all the implications of any changes I make. Meanwhile, I've added an errata item to highlight the issue for others and I will leave this issue open for now.
https://github.com/markjprice/cs11dotnet7/blob/main/docs/errata/errata.md#page-235---more-about-methods
